carp.common/src/commonMain/kotlin/dk/cachet/carp/common/application/devices/Website.kt

```diff
@@ -0,0 +1,105 @@
+@file:Suppress("NON_EXPORTABLE_TYPE")
+
+package dk.cachet.carp.common.application.devices
+
+import dk.cachet.carp.common.application.Trilean
+import dk.cachet.carp.common.application.data.DataType
+import dk.cachet.carp.common.application.sampling.DataTypeSamplingSchemeMap
+import dk.cachet.carp.common.application.sampling.SamplingConfiguration
+import dk.cachet.carp.common.application.tasks.TaskConfigurationList
+import dk.cachet.carp.common.application.toTrilean
+import dk.cachet.carp.common.infrastructure.serialization.NotSerializable
+import kotlinx.serialization.Required
+import kotlinx.serialization.Serializable
+import kotlin.js.JsExport
+import kotlin.reflect.KClass
+
+
+/**
+ * A device representing a website.
+ *
+ * @param unsupportedBrowsers A set of web [Browser]s which do not yield a valid [WebsiteDeviceRegistration].
+ * Empty by default.
+ */
+@Serializable
+@JsExport
+data class Website(
+    val unsupportedBrowsers: Set<Browser> = emptySet(),
+    override val roleName: String,
+    override val isOptional: Boolean = false
+) : PrimaryDeviceConfiguration<WebsiteDeviceRegistration, WebsiteDeviceRegistrationBuilder>()
+{
+    object Sensors : DataTypeSamplingSchemeMap()
+    object Tasks : TaskConfigurationList()
+
+    override fun getSupportedDataTypes(): Set<DataType> = Sensors.keys
+
+    override val defaultSamplingConfiguration: Map<DataType, SamplingConfiguration> = emptyMap()
+
+    override fun getDataTypeSamplingSchemes(): DataTypeSamplingSchemeMap = Sensors
+
+    override fun createDeviceRegistrationBuilder(): WebsiteDeviceRegistrationBuilder =
+            WebsiteDeviceRegistrationBuilder()
+
+    override fun getRegistrationClass(): KClass<WebsiteDeviceRegistration> = WebsiteDeviceRegistration::class
+
+    override fun isValidRegistration( registration: WebsiteDeviceRegistration ): Trilean =
+        unsupportedBrowsers.contains( registration.browser ).not().toTrilean()
+}
+
+/**
+ * Describes a browser type which can be used to make [WebsiteDeviceRegistration]s.
+ */
+@Serializable
+@JsExport
+enum class Browser
```

Comment:
This doesn't store the original value from the header, nor does the mapping.

Reply:
So, this is the core requirement of what we should be storing: https://en.wikipedia.org/wiki/User-Agent_header

I updated the PR accordingly.